tests/Io/RequestHeaderParserTest.php

```diff
@@ -831,6 +833,35 @@ public function testQueryParmetersWillBeSet()
         $this->assertEquals('this', $queryParams['test']);
     }
 
+    public function testIdleConnectionWillBeClosedAfterConfiguredTimeout()
+    {
+        $connection = $this->getMockBuilder('React\Socket\Connection')->disableOriginalConstructor()->setMethods(array('close'))->getMock();
+        $connection->expects($this->once())->method('close');
+
+        $parser = new RequestHeaderParser(Loop::get(), new Clock(Loop::get()), 0.1);
+
+        $parser->handle($connection);
+
+        $connection->emit('data', array("GET /foo.php?hello=world&test=this HTTP/"));
+
+        Loop::addTimer(0.2, function () {
+            Loop::stop();
+        });
+        Loop::run();
+    }
+
+    public function testIdleConnectionWillNotBeClosedAfterConfiguredTimeoutOnFullRequest()
+    {
+        $connection = $this->getMockBuilder('React\Socket\Connection')->disableOriginalConstructor()->setMethods(array('close'))->getMock();
+        $connection->expects($this->never())->method('close');
+
+        $parser = new RequestHeaderParser(Loop::get(), new Clock(Loop::get()), 0.1);
```

Comment:
This seems quite fragile, should we mock the loop here instead?

Reply:
Done